backend/src/datasources/postDatasource.js

```diff
@@ -62,13 +121,34 @@ export class PostDataSource extends DataSource {
       author: author
     })
 
+    const session = driver.session()
+    const txc = session.beginTransaction()
+    try {
+      await txc.run(
+        'CREATE (post:Post {id: $idParam, title: $titleParam, votes: 0, voters: []}) ' +
+        'WITH post MATCH (user:User) WHERE user.id = $userParam ' +
+        'CREATE (user) - [r:WROTE] -> (post)', {
+          idParam: newPost.id,
+          titleParam: args.post.title,
+          userParam: author.id
+        })
+
+      await txc.commit()
+    } catch (error) {
+      console.log(error)
+    } finally {
+      await session.close()
+    }
+
     author.posts.push(newPost)
     this.posts.push(newPost)
     return newPost
   }
```

Comment:
It would be better to ask the database about the post/author. If the database - for some reason - did not save your transaction properly (or changed it), the client wouldn't know as your pushing the post to `this.post` as well as `author.posts`.
You should not have a state in this class at all. 
`this.posts` should get removed from the class, it's constructor as well as everywhere else. 
I see that you already removed it most of the places. Maybe you just forgot the last 3 occurrences?